src/hevm/src/EVM.hs

```diff
@@ -1009,7 +1009,36 @@ exec1 = do
 
         -- op: REVERT
         0xfd ->
-          vmError Revert
+          case stk of
+            (xOffset:xSize:_) ->
+              accessMemoryRange fees xOffset xSize $ do
+                let output = readMemory (num xOffset) (num xSize) vm
+                case vm ^. frames of
+                  [] -> assign result (Just (VMFailure (Revert output)))
```

Comment:
One bit I wasn't totally sure about, whether a revert counts as a VMFailure or a VMSuccess. @mbrock?

Reply:
I think many implementations have a custom VMRevert status for revert. This is at least what KEVM and https://github.com/ethereum/evmjit/blob/gh-pages/libevmjit/JIT.h#L92 does

Reply:
In our case `VMFailure` is right because it means the transaction didn't succeed. The `VMResult` (which is either success or failure) is only used to denote the result when the whole "transaction" exits (when the VM terminates and the stack frame is empty).